crates/core/src/types.rs

```diff
@@ -10,11 +10,82 @@ use uuid::Uuid;
 /// Block time in microseconds
 pub const BLOCK_TIME: u128 = 2_000_000;
 
+#[derive(Default, Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct BundleTransactions(Vec<OpTxEnvelope>);
+
+impl From<Vec<OpTxEnvelope>> for BundleTransactions {
+    fn from(txs: Vec<OpTxEnvelope>) -> Self {
+        BundleTransactions(txs)
+    }
+}
+
+impl TryFrom<Vec<Bytes>> for BundleTransactions {
+    type Error = String;
+    fn try_from(txs: Vec<Bytes>) -> Result<Self, Self::Error> {
+        let transactions = txs
+            .iter()
+            .map(|b| OpTxEnvelope::decode_2718_exact(b))
+            .collect::<Result<Vec<OpTxEnvelope>, _>>();
+
+        match transactions {
+            Ok(transactions) => Ok(BundleTransactions(transactions)),
+            Err(e) => Err(format!("failed to decode transactions: {e}")),
+        }
+    }
+}
+
+impl From<BundleTransactions> for Vec<Bytes> {
+    fn from(txs: BundleTransactions) -> Self {
+        txs.0.iter().map(|t| t.encoded_2718().into()).collect()
+    }
+}
+
+impl BundleTransactions {
+    pub fn bundle_hash(&self) -> B256 {
+        let mut concatenated = Vec::new();
+        for tx in self.0.iter() {
+            concatenated.extend_from_slice(tx.tx_hash().as_slice());
+        }
+        keccak256(&concatenated)
+    }
+
+    /// Get transaction hashes for all transactions in the bundle
+    pub fn txn_hashes(&self) -> Vec<TxHash> {
+        self.0.iter().map(|t| t.tx_hash()).collect()
+    }
+
+    /// Get sender addresses for all transactions in the bundle
+    pub fn senders(&self) -> Vec<Address> {
+        self.0.iter().map(|t| t.recover_signer().unwrap()).collect()
+    }
+
+    /// Get total gas limit for all transactions in the bundle
+    pub fn gas_limit(&self) -> u64 {
+        self.0.iter().map(|t| t.gas_limit()).sum()
+    }
+
+    /// Get total data availability size for all transactions in the bundle
+    pub fn da_size(&self) -> u64 {
+        self.0
+            .iter()
+            .map(|t| tx_estimated_size_fjord_bytes(&t.encoded_2718()))
+            .sum()
+    }
+}
+
 #[derive(Default, Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
 #[serde(rename_all = "camelCase")]
 pub struct Bundle {
     pub txs: Vec<Bytes>,
 
+    #[serde(flatten)]
+    pub params: BundleParams,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct BundleParams {
     #[serde(with = "alloy_serde::quantity")]
     pub block_number: u64,
```

Comment:
What's the reasoning to change the structure of a `Bundle`? I believe this will change the API and cause it to diverge from `eth_sendBundle`.